src/utils/fs.js

```diff
@@ -11,6 +11,30 @@ const safeUnlink = async (path) => {
   } catch (_) {}
 }
 
+// Takes a list of absolute paths and returns an array containing all the
+// filenames within those directories, if at least one of the directories
+// exists. If not, an error is thrown.
+const listFunctionsDirectories = async function (srcFolders) {
+  const filenamesByDirectory = await Promise.all(
+    srcFolders.map(async (srcFolder) => {
+      try {
+        const filenames = await listFunctionsDirectory(srcFolder)
+
+        return filenames
+      } catch (error) {
```

Comment:
Instead of ignoring exceptions, should we check for `srcFolder` existence before running `listFunctionsDirectory()`?

Reply:
`listFunctionsDirectory` itself follows this logic of treating exceptions from `pReaddir` as signals that the directory doesn't exist, so we'll only ever fall into the `catch` branch here if one of the directories doesn't exist. Do you have a specific case in mind for also checking for the existence of the directory, which means another I/O call?

Reply:
Good point about the additional I/O call :+1: 
I am thinking of the case where `readdir` fails for another reason than `ENOENT`, e.g. `EPERM`. In that case, it might be good to propagate the error?

Reply:
Sorry for the late reply!

I see what you mean, but I think that's more about changing the existing `listFunctionsDirectory`, which is currently not propagating the error (or, more specifically, is swallowing the error message). Would you be okay with merging this as is and then figuring out how best to propagate errors in a subsequent PR?

Reply:
Sounds good :+1: